src-control/Control/RefCount.hs

```diff
@@ -496,7 +496,6 @@ assertNoUseAfterRelease Ref { reftracker = RefTracker refid _weak outer allocSit
         -- The site where the reference is used after release
         let useSite = callStack
         Control.Exception.throwIO (RefUseAfterRelease refid allocSite releaseSite useSite)
-    assertNoForgottenRefs
```

Comment:
I'm not sure I follow this.

How about keeping this, but dropping the `unsafePerformIO` in `deRef` instead?

Or otherwise distinguishing between the use of `assertNoForgottenRefs` in `dupRef` (monadic) vs in `reRef` (pure, cheating).

Reply:
> I'm not sure I follow this.
> 
> How about keeping this, but dropping the `unsafePerformIO` in `deRef` instead?

I don't think we can drop the `unsafePerformIO` if we want to run assertions in the `DeRef` pattern, right?

> Or otherwise distinguishing between the use of `assertNoForgottenRefs` in `dupRef` (monadic) vs in `reRef` (pure, cheating).

I'll do this